Sources/CodeEditTextView/EmphasisManager/EmphasisManager.swift

```diff
@@ -0,0 +1,362 @@
+    case outline(color: NSColor)
+
+    public static func == (lhs: EmphasisStyle, rhs: EmphasisStyle) -> Bool {
+        switch (lhs, rhs) {
+        case (.standard, .standard):
+            return true
+        case (.underline(let lhsColor), .underline(let rhsColor)):
+            return lhsColor == rhsColor
+        case (.outline(let lhsColor), .outline(let rhsColor)):
+            return lhsColor == rhsColor
+        default:
+            return false
+        }
+    }
+}
+
+/// Represents a single emphasis with its properties
+public struct Emphasis {
+    public let range: NSRange
+    public let style: EmphasisStyle
+    public let flash: Bool
+    public let inactive: Bool
+    public let select: Bool
+
+    public init(
+        range: NSRange,
+        style: EmphasisStyle = .standard,
+        flash: Bool = false,
+        inactive: Bool = false,
+        select: Bool = false
+    ) {
+        self.range = range
+        self.style = style
+        self.flash = flash
+        self.inactive = inactive
+        self.select = select
+    }
+}
+
+/// Manages text emphases within a text view, supporting multiple styles and groups.
+public final class EmphasisManager {
+    /// Internal representation of a emphasis layer with its associated text layer
+    private struct EmphasisLayer {
+        let emphasis: Emphasis
+        let layer: CAShapeLayer
+        let textLayer: CATextLayer?
+    }
+
+    private var emphasisGroups: [String: [EmphasisLayer]] = [:]
+    private let activeColor: NSColor = .findHighlightColor
+    private let inactiveColor: NSColor = NSColor.lightGray.withAlphaComponent(0.4)
+    private var originalSelectionColor: NSColor?
+
+    weak var textView: TextView?
+
+    init(textView: TextView) {
+        self.textView = textView
+    }
+
+    /// Adds a single emphasis to the specified group.
+    /// - Parameters:
+    ///   - emphasis: The emphasis to add
+    ///   - id: The group identifier
+    public func addEmphasis(_ emphasis: Emphasis, for id: String) {
+        addEmphases([emphasis], for: id)
+    }
+
+    /// Adds multiple emphases to the specified group.
+    /// - Parameters:
+    ///   - emphases: The emphases to add
+    ///   - id: The group identifier
+    public func addEmphases(_ emphases: [Emphasis], for id: String) {
+        // Store the current selection background color if not already stored
+        if originalSelectionColor == nil {
+            originalSelectionColor = textView?.selectionManager.selectionBackgroundColor ?? .selectedTextBackgroundColor
+        }
+
+        let layers = emphases.map { createEmphasisLayer(for: $0) }
+        emphasisGroups[id] = layers
+
+        // Handle selections
+        handleSelections(for: emphases)
+
+        // Handle flash animations
+        for (index, emphasis) in emphases.enumerated() where emphasis.flash {
+            let layer = layers[index]
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
```suggestion
            DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) { [weak self] in
                guard let self = self else { return }
```
This class may be deallocated while this waits (eg if the user manages to close the editor in that time) so this should weakly reference `self` to avoid a reference cycle.